node/overseer/overseer-gen/proc-macro/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+#![deny(unused_crate_dependencies)]
+
+use proc_macro2::TokenStream;
+use syn::{parse2, Result};
+
+mod impl_builder;
+mod impl_misc;
+mod impl_overseer;
+mod parse_attr;
+mod parse_struct;
+// mod impl_replace;
```

Comment:
is it not used?

Reply:
Currently `AllSubsystems` exists. This on the other hand will replace the `AllSubsystemsGen` proc-macro and work directly on the `#[overlord(..)]` generated struct and replace subsystems there as needed.

I will remove that part and re-introduce when ready.